darts/tests/test_timeseries_generation.py

```diff
@@ -12,56 +12,66 @@
 class TimeSeriesGenerationTestCase(unittest.TestCase):
 
     def test_constant_timeseries(self):
-
         # testing parameters
-        length = 100
         value = 5
 
-        # testing for constant value
-        constant_ts = constant_timeseries(value=value, length=length)
-        value_set = set(constant_ts._df.values.flatten())
-        self.assertTrue(len(value_set) == 1)
+        def test_routine(length):
+            # testing for constant value
+            constant_ts = constant_timeseries(value=value, length=length)
+            value_set = set(constant_ts._df.values.flatten())
+            self.assertTrue(len(value_set) == 1)
+
+        for length in range(1, 50):
+            test_routine(length)
 
     def test_linear_timeseries(self):
 
         # testing parameters
-        length = 100
         start_value = 5
         end_value = 12
 
-        # testing for start value, end value and delta between two adjacent entries
-        linear_ts = linear_timeseries(start_value=start_value, end_value=end_value, length=length)
-        self.assertEqual(linear_ts.values()[0][0], start_value)
-        self.assertEqual(linear_ts.values()[-1][0], end_value)
-        self.assertAlmostEqual(linear_ts.values()[-1][0] - linear_ts.values()[-2][0],
-                               (end_value - start_value) / (length - 1))
+        def test_routine(length):
+            # testing for start value, end value and delta between two adjacent entries
+            linear_ts = linear_timeseries(start_value=start_value, end_value=end_value, length=length)
+            self.assertEqual(linear_ts.values()[0][0], start_value)
+            self.assertEqual(linear_ts.values()[-1][0], end_value)
+            self.assertAlmostEqual(linear_ts.values()[-1][0] - linear_ts.values()[-2][0],
+                                   (end_value - start_value) / (length - 1))
+
+        for length in range(2, 50):
```

Comment:
seems to be doing much more iterations now, how much slower is it?

Reply:
Also - do we need to go over all interegs from 2 to 50 or is it enough to pick up start, middle and end point?

Reply:
I feel like we don't have to since this tests are really fast. But it is probably enough to test few options here so will change it to your proposition

Reply:
> I feel like we don't have to worry about it so much since this tests are really fast. But it is probably enough to test few options here so will change it to your proposition

Reply:
For the record github actions were around 30 seconds slower so in the end it was worth it to change it.